benchmarks/src/main/scala/benchmark/MergeDataLoad.scala

```diff
@@ -0,0 +1,197 @@
+ *                              condition.
+ */
+case class MergeSourceTable(
+    fileMatchedFraction: Double,
+    rowMatchedFraction: Double,
+    rowNotMatchedFraction: Double) {
+  def name: String = formatTableName(s"source_" +
+    s"_fileMatchedFraction_$fileMatchedFraction" +
+    s"_rowMatchedFraction_$rowMatchedFraction" +
+    s"_rowNotMatchedFraction_$rowNotMatchedFraction")
+
+  protected def formatTableName(s: String): String = {
+    s.toLowerCase(Locale.ROOT).replaceAll("\\s+", "_").replaceAll("[-,.]", "_")
+  }
+}
+
+object MergeDataLoadConf {
+  import scopt.OParser
+  private val builder = OParser.builder[MergeDataLoadConf]
+  private val argParser = {
+    import builder._
+    OParser.sequence(
+      programName("Merge Data Load"),
+      opt[String]("scale-in-gb")
+        .required()
+        .valueName("<scale of benchmark in GBs>")
+        .action((x, c) => c.copy(scaleInGB = x.toInt))
+        .text("Scale factor of the Merge benchmark"),
+      opt[String]("benchmark-path")
+        .required()
+        .valueName("<cloud storage path>")
+        .action((x, c) => c.copy(benchmarkPath = Some(x)))
+        .text("Cloud storage path to be used for creating table and generating reports"),
+      opt[String]("db-name")
+        .optional()
+        .valueName("<database name>")
+        .action((x, c) => c.copy(userDefinedDbName = Some(x)))
+        .text("Name of the target database to create with TPC-DS tables in necessary format"),
+      opt[String]("source-path")
+        .optional()
+        .valueName("<path to the TPC-DS raw input data>")
+        .action((x, c) => c.copy(sourcePath = Some(x)))
+        .text("The location of the TPC-DS raw input data"),
+      opt[String]("exclude-nulls")
+        .optional()
+        .valueName("true/false")
+        .action((x, c) => c.copy(excludeNulls = x.toBoolean))
+        .text("Whether to remove null primary keys when loading data, default = false"),                                                                                                                                                          )
+  }
+
+  def parse(args: Array[String]): Option[MergeDataLoadConf] = {
+    OParser.parse(argParser, args, MergeDataLoadConf())
+  }
+}
+
+class MergeDataLoad(conf: MergeDataLoadConf) extends Benchmark(conf) {
+
+  protected def targetTableFullName = s"`${conf.dbName}`.`target_${conf.tableName}`"
+
+  protected def dataSourceLocation: String = conf.sourcePath.getOrElse {
+    s"s3://devrel-delta-datasets/tpcds-2.13/tpcds_sf${conf.scaleInGB}_parquet/${conf.tableName}/"
+  }
+
+  /**
+   * Creates the target table and all source table configuration used in merge test cases.
+   */
+  def runInternal(): Unit = {
+    val dbName = conf.dbName
+    val dbLocation = conf.dbLocation(dbName, suffix = benchmarkId.replace("-", "_"))
+    val dbCatalog = "spark_catalog"
+
+    require(conf.scaleInGB > 0)
```

Comment:
Do you need this when you have the next line?

Reply:
No, removed